pkg/monitortests/node/watchnodes/node.go

```diff
@@ -157,6 +148,48 @@ func startNodeMonitoring(ctx context.Context, m monitorapi.RecorderWriter, clien
 			}
 			return intervals
 		},
+		// This function is added to help detect unexpected
+		// node not ready.
+		// We want to fail the monitor test if a node goes not ready
+		// if it is unexpected.
+		// Unexpected in this case means that it went not ready outside
+		// of a MCO config update.
+		func(node, oldNode *corev1.Node) []monitorapi.Interval {
+			var intervals []monitorapi.Interval
+
+			oldConfig := oldNode.Annotations["machineconfiguration.openshift.io/currentConfig"]
+			newConfig := node.Annotations["machineconfiguration.openshift.io/currentConfig"]
+			isOldNodeReady := isNodeReady(oldNode)
+			isNewNodeReady := isNodeReady(node)
+			isConfigTheSame := oldConfig == newConfig
+
+			now := time.Now()
+			if isOldNodeReady && !isNewNodeReady && isConfigTheSame {
+				intervals = append(intervals,
+					monitorapi.NewInterval(monitorapi.SourceNodeMonitor, monitorapi.Error).
```

Comment:
Suggest adding a new Source per reason essentially, and make sure to call .Display() before the Build() call. 

The new sippy UI should automatically pick that up as an interval category, you should then see it in the row of boxes at the top of the chart, and can enable it quickly from there.

Source per type of interval will avoid potential overlap where if the node locator is identical, and two things overlap, one could be hidden as the other renders over it. We hack around this in NodeState with the "row" you see in the locator on the right. But if you do the above Source per type of interval you are generating (looks like two here), this should not be an issue.

So essentially just change source here to something new, monitorapi.SourceNodeUnexpectedNotReady, and monitorapi.SourceNodeUnreachable, those both should then automatically appear as checkboxes in the sippy intervals UI. (remembering to set Display() on both)

Reply:
Updated.